xcm/src/v1/multilocation.rs

```diff
@@ -36,83 +39,385 @@ use parity_scale_codec::{self, Decode, Encode};
+	}
+
+	/// Mutates `self`, suffixing its interior junctions with `new`. Returns `Err` in case of overflow.
+	pub fn push_interior(&mut self, new: Junction) -> result::Result<(), ()> {
+		let mut n = Junctions::Null;
+		mem::swap(&mut self.interior, &mut n);
+		match n.pushed_with(new) {
+			Ok(result) => {
+				self.interior = result;
+				Ok(())
+			},
+			Err(old) => {
+				self.interior = old;
+				Err(())
+			},
+		}
+	}
+
+	/// Mutates `self`, prefixing its interior junctions with `new`. Returns `Err` in case of overflow.
+	pub fn push_front_interior(&mut self, new: Junction) -> result::Result<(), ()> {
+		let mut n = Junctions::Null;
+		mem::swap(&mut self.interior, &mut n);
+		match n.pushed_front_with(new) {
+			Ok(result) => {
+				self.interior = result;
+				Ok(())
+			},
+			Err(old) => {
+				self.interior = old;
+				Err(())
+			},
+		}
+	}
+
+	/// Consumes `self` and returns a `MultiLocation` with its parent count incremented by 1, or
+	/// an `Err` with the original value of `self` in case of overflow.
+	pub fn pushed_with_parent(self) -> result::Result<Self, Self> {
+		if self.len() >= MAX_MULTILOCATION_LENGTH {
+			return Err(self)
+		}
+		Ok(MultiLocation { parents: self.parents.saturating_add(1), ..self })
+	}
+
+	/// Consumes `self` and returns a `MultiLocation` suffixed with `new`, or an `Err` with the original value of
+	/// `self` in case of overflow.
+	pub fn pushed_with_interior(self, new: Junction) -> result::Result<Self, Self> {
+		if self.len() >= MAX_MULTILOCATION_LENGTH {
+			return Err(self)
+		}
+		Ok(MultiLocation {
+			parents: self.parents,
+			interior: self.interior.pushed_with(new).expect("length is less than max length; qed"),
+		})
+	}
+
+	/// Consumes `self` and returns a `MultiLocation` prefixed with `new`, or an `Err` with the original value of
+	/// `self` in case of overflow.
+	pub fn pushed_front_with_interior(self, new: Junction) -> result::Result<Self, Self> {
+		if self.len() >= MAX_MULTILOCATION_LENGTH {
+			return Err(self)
+		}
+		Ok(MultiLocation {
+			parents: self.parents,
+			interior: self
+				.interior
+				.pushed_front_with(new)
+				.expect("length is less than max length; qed"),
+		})
+	}
+
+	/// Returns the junction at index `i`, or `None` if the location is a parent or if the location
+	/// does not contain that many elements.
+	pub fn at(&self, i: usize) -> Option<&Junction> {
+		let num_parents = self.parents as usize;
+		if i < num_parents {
+			return None
+		}
+		self.interior.at(i - num_parents)
+	}
+
+	/// Returns a mutable reference to the junction at index `i`, or `None` if the location is a
+	/// parent or if it doesn't contain that many elements.
+	pub fn at_mut(&mut self, i: usize) -> Option<&mut Junction> {
+		let num_parents = self.parents as usize;
+		if i < num_parents {
+			return None
+		}
+		self.interior.at_mut(i - num_parents)
+	}
+
+	/// Decrements the parent count by 1.
+	pub fn pop_parent(&mut self) {
+		self.parents = self.parents.saturating_sub(1);
```

Comment:
```suggestion
		self.parents = self.parents.saturating_dec();
```

Reply:
Can't, `saturating_dec` is only defined in `frame_support`, and we're not going to import that here in `xcm`.